blog/09-05-2023-smart-contract-security-on-celo-with-securify/smart-contract-security-on-celo-with-securify.md

```diff
@@ -0,0 +1,276 @@
+
+```bash
+solc-select install
+```
+In my terminal, I got several previously installed versions.
+
+![image](images/1.png)
+
+But if nothing was returned in your terminal, you will have to install the versions you want. You can install a specific version or simply get the latest version. 
+
+```bash
+    solc-select install <version>
+```
+
+or 
+
+```bash
+    solc-select latest
+```
+
+After installation, we have to explicitly tell `solc` which version we are using. 
+
+```bash
+    solc-select use <version>
+```
+
+Example: 
+
+```bash
+    solc-select use 0.8.18
+```
+
+or simply use the latest version.
+
+```bash
+    solc-select use latest
+```
+
+Now that we have instructed `solc-select` which version to use, let's make Slither invoke the compiler. We have to create a project first.
+
+**Project setup**
+
+- Create a project folder
+
+```bash
+    mkdir <projectName> && cd <projectName>
+```
+
+- Create hardhat project
+
+```bash Yarn
+    yarn add hardhat
+```
+```bash NPM
+    npm install hardhat
+```
+
+- Invoke hardhat
+
+```bash
+    npx hardhat
+```
+
+Then, follow the instruction to complete the installation.
+
+We will use the boilerplate code provided to us by Hardhat. Create a new file, name it `lock2.sol`. Populate it with the code below.
+
+```js Lock2.sol
+    // SPDX-License-Identifier: UNLICENSED
+    pragma solidity ^0.8.9;
+
+    contract Lock {
+        uint public unlockTime;
+
+        mapping(address => uint256) public balances;
+
+        event Withdrawal(uint amount, uint when);
+
+        constructor(uint _unlockTime) payable {
+            require(
+                block.timestamp < _unlockTime,
+                "Unlock time should be in the future"
+            );
+
+            unlockTime = _unlockTime;
+        }
+
+        function withdraw(uint amount) public {
+            // Uncomment this line, and the import of "hardhat/console.sol", to print a log in your terminal
+            // console.log("Unlock time is %o and block timestamp is %o", unlockTime, block.timestamp);
+            uint bal = balances[msg.sender];
+            require(block.timestamp >= unlockTime, "You can't withdraw yet");
+            require(amount > 0, "Nothing to withdraw");
+            (bool sent,) = address(msg.sender).call{value: amount}("");
+            require(sent);
+            emit Withdrawal(address(this).balance, block.timestamp);
+
+            balances[msg.sender] -= amount;
+        }
+    }
+
+```
+
+>Note: I only modified the existing content for tutorial purposes. Do not use it in production.
+
+Let's ask Slither to review the current status of the code. Run the following command by invoking Slither followed by the path to the contract.
+
+```bash
+    slither contracts/Lock2.sol
+```
+
+![image](images/2.png)
```

Comment:
the image you mean is not in your tutorial directory

Reply:
> the image you mean is not in your tutorial directory

I think you're reviewing the wrong content. Please check again